### PR TITLE
Select labs

### PR DESCRIPTION
**Pull request by johnagl**
# Changes
- Labs screen displays schedule that was selected from Lectures screen, Generate screen displays lectures with labs

- **There are a number of missing features at the moment, so this isn't 100% complete, some of the following things I can think of are: **
  - We can't delete a lab section after adding it
  - User can add many lab sections for the same course 
  - There are no checks for overlapping times
  - The current implementation re-adds the entire schedules array into redux store (bad) 
  - Overall design isnt great and I'll probably refactor it during winter break
  - I didn't style it at all

## Motivation and Context
- After user selects a schedule on lectures screen, we show them the labs screen where they can add lab and other non-lecture sections

## How Has This Been Tested?
- Manually

## Screenshots (if appropriate):
selectLabs

# Related Issues
- Closes issue #

# Checklist

### CODE REVIEW

File: client/src/components/ScheduleGrid.tsx [hunk 1]
```diff
@@ -51,20 +55,51 @@ const useStyles = makeStyles({
...
+            const [eh, em] = endtime.split(":");
+            const start = Number(sh) + (Number(sm) / 60);
+            const end = Number(eh) + (Number(em) / 60);
+            const days = day.split(" ");
+            days.forEach((dayOfWeek: any) => addCourse(rows, dayOfWeek, start, end, sectiontitle, i));
+        });
+    } else {
+        if (selectedSchedule && schedules) {
+            // TODO: dont repeat this code
+            schedules[selectedSchedule].forEach((section: any, i: number) => {
```
Comment (yehee) [inline client/src/components/ScheduleGrid.tsx:92]: What's the reason for this else case?

Comment (johnagl) [reply client/src/components/ScheduleGrid.tsx:92]: In Labs/Generate we call <Schedule><Schedule/> component without passing it a schedule prop. We grab the schedule from the redux store. This design isn't great atm - but only way I could think of doing it atm.

Comment (johnagl) [reply client/src/components/ScheduleGrid.tsx:92]: I also didn't want to remake ScheduleGrid component for Labs/Generate (wanted to reuse what we currently have).

Comment (johnagl) [reply client/src/components/ScheduleGrid.tsx:92]: We can probably set Redux schedule state with the selected course schedule, and pass that into ScheduleGrid so we don't need the else - but I just thought of that now :D

File: client/src/components/Labs.tsx [hunk 2]
```diff
@@ -2,15 +2,65 @@ import React from "react";
...
+  selectedSchedule: number;
+  schedules: CourseSection[][];
+  notLectureSections: CourseSection[];
+  addSchedules?: any;
+}
+
+
+function Labs({selectedSchedule, schedules, notLectureSections, addSchedules}: LabsProps) {
+  const handleClick = (section: CourseSection) => {
+    // TODO: This isnt a good way to add lectures, since we're replacing the entire schedules state in the store
```
Comment (johnagl) [inline client/src/components/Labs.tsx:23]: ** meant to say this isnt a good way to add non lectures

**Comment by yehee** [comment]
@johnagl Testing it locally and right now I don't see the schedule getting populated on Add Lab Sections page, was wondering if it's functioning properly for others?

**Comment by johnagl** [comment]
> @johnagl Testing it locally and right now I don't see the schedule getting populated on Add Lab Sections page, was wondering if it's functioning properly for others?

Ty for fixing it :)

**Pull request merged**